ibis/bigquery/client.py

```diff
@@ -0,0 +1,207 @@
+
+    @property
+    def _context(self):
+        return _bq_get_context(self._project_id)
+
+    @property
+    def _dataset(self):
+        return _bq_get_dataset(self._dataset_id, self._project_id)
+
+    def _build_ast(self, expr, params=None):
+        return comp.build_ast(expr, params=params)
+
+    def execute(self, expr, limit='default', async=False, params=None,
+                output_options=None):
+        if limit != 'default' or async or params:
+            raise NotImplementedError()
+
+        stmt = expr.compile()
+        return (bq.Query(stmt)
+                .execute(output_options=output_options, context=self._context)
+                .result()
+                .to_dataframe()
+                )
+
+    @property
+    def _table_expr_klass(self):
+        return ir.TableExpr
+
+    def _fully_qualified_name(self, table_id, dataset_id=None):
+        dataset_id = dataset_id or self._dataset_id
+        return dataset_id + '.' + table_id
+
+    def _get_table_schema(self, qualified_name):
+        return self.get_schema(qualified_name)
+
+    def list_tables(self, like=None, dataset=None):
+        dataset = _bq_get_dataset(dataset or self._dataset_id,
+                                  self._project_id)
+        result = [table.name.table_id for table in dataset.tables()]
+        if like:
+            result = [table_name
+                      for table_name in result if re.match(like, table_name)]
+        return result
+
+    def set_dataset(self, name):
+        self.__dataset_id = name
+
+    def exists_database(self, name):
+        return _bq_get_dataset(name, self._project_id).exists()
+
+    def list_datasets(self, like=None):
+        results = [dataset.name.dataset_id
+                   for dataset in _bq_get_datasets(self._project_id)]
+        if like:
+            results = [dataset_name
+                       for dataset_name in results
+                       if re.match(like, dataset_name)
+                       ]
+        return results
+
+    def get_schema(self, name, dataset=None):
+        (table_id, dataset_id) = _ensure_split(name, dataset)
+        if dataset_id and dataset:
+            raise ValueError(
+                'Can\'t pass a fully qualified table name *AND* a dataset'
+            )
+        else:
+            dataset_id = dataset or self._dataset_id
+        table = _bq_get_table(table_id, dataset_id, self._project_id)
+        return bigquery_dtypes_to_ibis_schema(table, None)
+
+    def exists_table(self, name, dataset=None):
+        dataset_id = dataset or self._dataset_id
+        return _bq_get_table(name, dataset_id, self._project_id).exists()
+
+
+_DTYPE_TO_IBIS_TYPE = {
+    'INT64': dt.int64,
+    'FLOAT64': dt.double,
+    'BOOL': dt.boolean,
+    'STRING': dt.string,
+    'DATE': dt.date,
+    # FIXME: enforce no tz info
+    'DATETIME': dt.timestamp,
+    'TIME': dt.time,
+    'TIMESTAMP': dt.timestamp,
+    # 'BYTES': None,
+    # 'ARRAY': None,
+    # 'STRUCT': None,
```

Comment:
Do you need any of these now that you have an implementation that can handle arrays and structs? What about `BYTES`?